src/mozanalysis/experiment.py

```diff
@@ -679,6 +679,8 @@ class TimeLimits(object):
     first_date_data_required = attr.ib(type=str)
     last_date_data_required = attr.ib(type=str)
 
+    analysis_windows = attr.ib(type=tuple)
```

Comment:
`type=Iteratable[AnalysisWindow]`?

Reply:
I don't think that type checking here is worth dropping python 2 support or doing version detection, especially since this is an internal object. Also it's a tuple, for immutability.

Reply:
I'm not arguing for type checking, I don't think. Assigning `type=` is a no-op, anyway, and only serves for documentation; `tuple` on its own isn't very good documentation.

Reply:
Hmm. Would you prefer
```    analysis_windows = attr.ib(type=tuple)  # Tuple of AnalysisWindows```

Reply:
Or
```analysis_windows = attr.ib()  # type: tuple[AnalysisWindow]```